backend/src/main/java/org/cryptomator/hub/api/UsersResource.java

```diff
@@ -156,10 +160,23 @@ public Response updateMyAccessTokens(@NotNull Map<UUID, String> tokens) {
 	@Operation(summary = "get the logged-in user")
 	@APIResponse(responseCode = "200", description = "returns the current user")
 	@APIResponse(responseCode = "404", description = "no user matching the subject of the JWT passed as Bearer Token")
-	public UserDto getMe(@QueryParam("withDevices") boolean withDevices) {
+	public UserDto getMe(@QueryParam("withDevices") boolean withDevices, @QueryParam("withLastAccess") boolean withLastAccess) {
 		User user = userRepo.findById(jwt.getSubject());
-		Function<Device, DeviceResource.DeviceDto> mapDevices = d -> new DeviceResource.DeviceDto(d.getId(), d.getName(), d.getType(), d.getPublickey(), d.getUserPrivateKeys(), d.getOwner().getId(), d.getCreationTime().truncatedTo(ChronoUnit.MILLIS));
-		var devices = withDevices ? user.devices.stream().map(mapDevices).collect(Collectors.toSet()) : Set.<DeviceResource.DeviceDto>of();
+		Set<DeviceResource.DeviceDto> devices;
+		if (withLastAccess) {
+			var deviceEntities = user.devices.stream().toList();
+			var deviceIds = deviceEntities.stream().map(Device::getId).toList();
+			var events = auditEventRepo.findLastVaultKeyRetrieve(deviceIds).collect(Collectors.toMap(VaultKeyRetrievedEvent::getDeviceId, Function.identity()));
+			devices = deviceEntities.stream().map(d -> {
+				var event = events.get(d.getId());
+				var lastIpAddress = (event != null) ? event.getIpAddress() : null;
+				var lastAccessTime = (event != null) ? event.getTimestamp() : null;
+				return new DeviceResource.DeviceDto(d.getId(), d.getName(), d.getType(), d.getPublickey(), d.getUserPrivateKeys(), d.getOwner().getId(), d.getCreationTime().truncatedTo(ChronoUnit.MILLIS), lastIpAddress, lastAccessTime);
+			}).collect(Collectors.toSet());
```

Comment:
just as an alternative, what do you think about this?

```suggestion
			devices = deviceEntities.stream().map(d -> {
				var event = events.get(d.getId());
				return DeviceResource.fromEntity(d, event);
			}).collect(Collectors.toSet());
```

Adding `public static DeviceDto fromEntity(Device entity, @Nullable VaultKeyRetrievedEvent lastUnlock)`

Reply:
I really like it! Fixed in [9d448b7](https://github.com/cryptomator/hub/pull/320/commits/9d448b72c82f3b3e4de35036df3b8d3e075b6e8e)